service/frontend/workflowHandler_test.go

```diff
@@ -636,6 +636,93 @@ func (s *workflowHandlerSuite) TestRecordActivityTaskHeartbeat_TaskTokenNotSet()
 	s.Nil(result)
 }
 
+func (s *workflowHandlerSuite) TestRecordActivityTaskHeartbeatByID_Success() {
+	wh := s.getWorkflowHandler(s.newConfig(dc.NewInMemoryClient()))
+	req := &types.RecordActivityTaskHeartbeatByIDRequest{
+		Domain:     s.testDomain,
+		WorkflowID: testWorkflowID,
+		RunID:      testRunID,
+		ActivityID: "1",
+		Details:    nil,
+		Identity:   "",
+	}
+	resp := &types.RecordActivityTaskHeartbeatResponse{CancelRequested: false}
+
+	s.mockDomainCache.EXPECT().GetDomainID(s.testDomain).Return(s.testDomainID, nil)
+	s.mockHistoryClient.EXPECT().RecordActivityTaskHeartbeat(gomock.Any(), gomock.Any()).Return(resp, nil)
+
+	result, err := wh.RecordActivityTaskHeartbeatByID(context.Background(), req)
+	s.NoError(err)
+	s.Equal(resp, result)
+}
+
+func (s *workflowHandlerSuite) TestRecordActivityTaskHeartbeatByID_RequestNotSet() {
+	wh := s.getWorkflowHandler(s.newConfig(dc.NewInMemoryClient()))
+	result, err := wh.RecordActivityTaskHeartbeatByID(context.Background(), nil /*request is not set*/)
+
+	s.Error(err)
+	s.Equal(errRequestNotSet, err)
+	s.Nil(result)
+}
+
+func (s *workflowHandlerSuite) TestRecordActivityTaskHeartbeatByID_DomainNotSet() {
+	wh := s.getWorkflowHandler(s.newConfig(dc.NewInMemoryClient()))
+	result, err := wh.RecordActivityTaskHeartbeatByID(
+		context.Background(),
+		&types.RecordActivityTaskHeartbeatByIDRequest{
+			Domain: "", // domain not set
+		})
+
+	s.Error(err)
+	s.Equal(errDomainNotSet, err)
+	s.Nil(result)
+}
+
+func (s *workflowHandlerSuite) TestRespondActivityTaskCompleted_Success() {
+	wh := s.getWorkflowHandler(s.newConfig(dc.NewInMemoryClient()))
+	taskToken := common.TaskToken{
+		DomainID:   s.testDomainID,
+		WorkflowID: testWorkflowID,
+		RunID:      testRunID,
+		ActivityID: "1",
+	}
+	taskTokenBytes, err := wh.tokenSerializer.Serialize(&taskToken)
+	s.NoError(err)
+	req := &types.RespondActivityTaskCompletedRequest{
+		TaskToken: taskTokenBytes,
+		Result:    nil,
+		Identity:  "",
+	}
+
+	s.mockDomainCache.EXPECT().GetDomainName(s.testDomainID).Return(s.testDomain, nil)
+	s.mockHistoryClient.EXPECT().RespondActivityTaskCompleted(gomock.Any(),
+		&types.HistoryRespondActivityTaskCompletedRequest{
+			DomainUUID:      taskToken.DomainID,
+			CompleteRequest: req,
+		}).Return(nil)
+
+	err = wh.RespondActivityTaskCompleted(context.Background(), req)
```

Comment:
I'm surprised there is no return value or anything? Is there a way to check data is written?
nit: If it's not the point of test, I suggest to add a comment like "we're only checking for successful write here".